server/src/routes/for-progress2/reservation-list.ts

```diff
@@ -0,0 +1,37 @@
+import { sql } from '@/libs/db';
+import Elysia from 'elysia';
+
+export const reservationListRoute = new Elysia({ prefix: '/reservation-list' }).get(
+    '/',
+    async ({ set }) => {
+        const queryReservationList = await sql`SELECT
+	*
+FROM
+	reservations
+	INNER JOIN
+	customer_details
+	ON 
+		reservations.customer_id = customer_details."id"
+	INNER JOIN
+	rooms
+	ON 
+		reservations.room_id = rooms."id"
+	INNER JOIN
+	room_types
+	ON 
+		rooms.type_id = room_types."id"`;
+
+        if (queryReservationList.count === 0) {
```

Comment:
use `destructuring assignment` array to check if the object is whether `undefined` or not

```
const [queryReservationList] = await sql``;

if (!queryReservationList) {
   error;
}
```
under the hood is 
const [ A, B ] = [5, 6]
is equal to 
const A = 5;
const B = 6;

Reply:
! undefind = true 

so It will return `error`;